cognite/neat/core/_rules/importers/_yaml2rules.py

```diff
@@ -6,16 +6,38 @@
 from cognite.neat.core._issues import IssueList, MultiValueError, NeatIssue
 from cognite.neat.core._issues.errors import (
     FileMissingRequiredFieldError,
-    FileNotAFileError,
-    FileNotFoundNeatError,
     FileReadError,
-    FileTypeUnexpectedError,
 )
 from cognite.neat.core._issues.warnings import NeatValueWarning
 from cognite.neat.core._rules._shared import ReadRules, T_InputRules
 from cognite.neat.core._rules.models import INPUT_RULES_BY_ROLE, RoleTypes
 
 from ._base import BaseImporter
+from ._base_file_reader import BaseFileReader
+
+
+class YAMLReader:
+    """Handles reading and parsing YAML files with error handling."""
+
+    @staticmethod
+    def read_file(
+        filepath: Path, allowed_extensions: frozenset[str] = frozenset([".yaml", ".yml"])
+    ) -> tuple[dict[str, Any], list[NeatIssue]]:
+        """Read a YAML file and return the data with any issues encountered."""
+        issues = BaseFileReader.validate_file(filepath, allowed_extensions)
+        if issues:
+            return {}, issues
+        # Try to load the YAML
+        try:
+            data = yaml.safe_load(filepath.read_text())
+            if not isinstance(data, dict):
```

Comment:
An empty file will give you None back from safe_load, so you could special case that to specify that it is because of an empty file instead of "is not a dictionary").